### PR TITLE
Feature/upgrade to react 16.3.3

### PR DESCRIPTION
**Pull request by maoznir**

### CODE REVIEW

File: src/Util/cloudinaryProps.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import {Util} from "cloudinary-core";
+
+/**
+ * Returns a camelCase version of given props, filtered using cloudinaryOptions
```
Comment (asisayag2) [inline src/Util/cloudinaryProps.js]: @maoznir , don't you find the need to run on this list twice, once to get the cloudinary props and once to get the non-cloudinary props a bit too much? 
Can't we have a single run that separates the cloudinary from non-cloudinary props?

File: src/Util/nonCloudinaryProps.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+export default (props, cloudinaryProps) => {
```
Comment (asisayag2) [inline src/Util/nonCloudinaryProps.js]: Missing doc string

Comment (asisayag2) [inline src/Util/nonCloudinaryProps.js]: This is called cloudinaryOptions in the cloudinaryProps util function

File: src/components/CloudinaryContext/CloudinaryContext.js [hunk 3]
```diff
@@ -24,35 +15,32 @@ const getNonCloudinaryProps = (props) => {
...
-
-    // only pass valid Cloudinary options
-    CloudinaryComponent.VALID_OPTIONS.forEach(key => {
-      let val = camelProps[key] || this.context[key];
-      if (val !== undefined && val !== null) {
-        context[key] = val;
-      }
-    });
-    return context;
+    const mergedProps = {...(this.context || {}), ...this.props};
```
Comment (asisayag2) [inline src/components/CloudinaryContext/CloudinaryContext.js]: ??

File: src/components/CloudinaryContext/CloudinaryContext.js [hunk 4]
```diff
@@ -24,35 +15,32 @@ const getNonCloudinaryProps = (props) => {
...
+    const context = cloudinaryProps(mergedProps, CloudinaryComponent.VALID_OPTIONS);
```
Comment (asisayag2) [inline src/components/CloudinaryContext/CloudinaryContext.js]: Would call this method "extractCloudinaryProps"

File: src/components/CloudinaryContext/CloudinaryContext.js [hunk 5]
```diff
@@ -24,35 +15,32 @@ const getNonCloudinaryProps = (props) => {
...
+    const {includeOwnBody, ...childrenProps} = nonCloudinaryProps(this.props, CloudinaryContext.CLOUDINARY_PROPS);
```
Comment (asisayag2) [inline src/components/CloudinaryContext/CloudinaryContext.js]: Again, will have a single util to separate the cloudinary from the non-cloudinary props

File: src/components/CloudinaryContext/CloudinaryContext.js [hunk 6]
```diff
@@ -24,35 +15,32 @@ const getNonCloudinaryProps = (props) => {
...
-  constructor(props) {
-    super(props);
+  constructor(props, context) {
+    super(props, context);
+
     this.calcState = this.calcState.bind(this);
     this.state = this.calcState();
   }
 
   calcState() {
```
Comment (asisayag2) [inline src/components/CloudinaryContext/CloudinaryContext.js]: Please add a comment explaining what this does

File: src/components/CloudinaryContext/CloudinaryContext.js [hunk 7]
```diff
@@ -24,35 +15,32 @@ const getNonCloudinaryProps = (props) => {
...
-      if (val !== undefined && val !== null) {
-        context[key] = val;
-      }
-    });
-    return context;
+    const mergedProps = {...(this.context || {}), ...this.props};
+    const context = cloudinaryProps(mergedProps, CloudinaryComponent.VALID_OPTIONS);
+    const {includeOwnBody, ...childrenProps} = nonCloudinaryProps(this.props, CloudinaryContext.CLOUDINARY_PROPS);
+
+    return {
```
Comment (asisayag2) [inline src/components/CloudinaryContext/CloudinaryContext.js]: Why do we return the state and not set it?

File: test/setProps.js [hunk 8]
```diff
@@ -0,0 +1,14 @@
+/**
+ * Promisified enzyme function setProps()
+ * Sets the props of given react component
+ * @param component - enzyme
```
Comment (asisayag2) [inline test/setProps.js]: Are you using this util function in several files?

Comment (maoznir) [reply test/setProps.js]: yes, TransformationTest, ContextTest

File: src/components/CloudinaryContext/CloudinaryContext.js (hunk 6 above)
Comment (asisayag2) [reply src/components/CloudinaryContext/CloudinaryContext.js]: Still requires a comment

File: src/Util/extractCloudinaryProps.js [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+
+/**
+ * Extracts cloudinaryProps and nonCloudinaryProps from given context+props
+ * according to given validProps
+ * @param context
+ * @param props
+ * @param validProps
+ * @returns {{cloudinaryProps, nonCloudinaryProps}}
+ */
+export default (context = {}, props = {}, validProps = {}) => {
```
Comment (asisayag2) [inline src/Util/extractCloudinaryProps.js]: The merge does not belong here. 
It should be done outside and here you should get a list of props and a list of cloudinaryProps

**Pull request merged**